### PR TITLE
add automation to clean supporting hosted cluster secrets on delete

### PR DESCRIPTION
**Pull request by relyt0925**
fixes: https://github.com/openshift/hypershift/issues/147

### CODE REVIEW

File: hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go [hunk 1]
```diff
@@ -1498,6 +1498,19 @@ func (r *HostedClusterReconciler) delete(ctx context.Context, req ctrl.Request,
 	if err := r.Delete(ctx, ns); err != nil && !apierrors.IsNotFound(err) {
 		return false, fmt.Errorf("failed to delete namespace: %w", err)
 	}
+
+	secretSuffixesToRemove := []string{"pull-secret", "provider-creds", "signing-key", "ssh-key", "admin-kubeconfig"}
```
Comment (enxebre) [inline hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go:1502]: Other than the generated `admin-kubeconfig` these resources are not managed by `hostedcluster_controller`, they need to be created ahead of time and referenced by the hostedCluster API, it just happen that our dev cli automate their creation for convenience. So it is expected for them to be also deleted out of band. Do you agree?

Comment (enxebre) [reply hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go:1502]: For the `admin-kubeconfig` we can possibly enforce an `owner reference` during creation so it just gets garbage collected once it goes orphan when deleting the `hostedCluster`

Comment (relyt0925) [reply hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go:1502]: I'm totally cool with that! Just wasn't sure what the general strategy was there!

Comment (relyt0925) [reply hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go:1502]: I'll close since that makes sense

**Pull request closed without merging**